src/core/website_manager.py

```diff
@@ -1,311 +1,218 @@
-        headers = {
-            header_element["name"]: [header_element["value"].replace('"', "")]
-            for header_element in raw_headers
-        }
-        return headers
+        Will raise if splash is not reachable or the returned response is malformed (doesn't match expected structure).
+        """
 
-    def _get_html_and_har(self, url: str) -> dict:
         splash_url = (
-            f"{SPLASH_URL}/render.json?url={url}&html={1}&iframes={1}"
+            f"{SPLASH_URL}/render.json?url={message.url}&html={1}&iframes={1}"
             f"&har={1}&response_body={1}&wait={10}&render_all={1}&script={1}&timeout={SPLASH_TIMEOUT}"
         )
-        try:
-            response = requests.get(
-                url=splash_url,
-                headers=SPLASH_HEADERS,
-                params={},
+        response = requests.get(
+            url=splash_url,
+            headers=SPLASH_HEADERS,
+            params={},
+        )
+        data = json.loads(response.content.decode("UTF-8"))
+
+        return Message(
+            url=message.url,
+            html=data["html"],
+            header=data["har"]["log"]["entries"][0]["response"]["headers"],
+            har=data["har"],
+            extractors=message.extractors,
+            bypass_cache=message.bypass_cache,
+            _shared_memory_name=message._shared_memory_name,
+        )
+
+    @classmethod
+    def from_message(
+        cls, message: Message, tld_extractor: TLDExtract, logger: Logger
+    ) -> "WebsiteData":
+        def top_level_domain(url: str) -> str:
+            host = tld_extractor(
+                url=url.replace("http://", "").replace("https://", "")
             )
-            data = json.loads(response.content.decode("UTF-8"))
-        except (JSONDecodeError, OSError) as err:
-            exception = (
-                "Error extracting data from splash: "
-                + str(err.args)
-                + "".join(
-                    traceback.format_exception(None, err, err.__traceback__)
-                )
+            domain = [host.subdomain, host.domain, host.suffix]
+            return ".".join([element for element in domain if element != ""])
+
+        def extract_raw_links(
+            soup: BeautifulSoup, image_links: list[str]
+        ) -> list[str]:
+            source_regex = re.compile(
+                r"src\=[\"|\']([\w\d\:\/\.\-\?\=]+)[\"|\']"
             )
-            self._logger.error(exception)
-            data = {}
-        except Exception as err:
-            raise ConnectionError from err
-
-        try:
-            raw_headers = data["har"]["log"]["entries"][0]["response"][
-                "headers"
+            logger.debug("extracting raw links")
+            unique_tags = get_unique_list(
+                [tag.name for tag in soup.find_all()]
+            )
+            logger.debug(f"unique_tags: {unique_tags}")
+            if SCRIPT in unique_tags:
+                unique_tags.remove(SCRIPT)
```

Comment:
Why do you remove the tag here? Is it because of later usage on line 112?

Reply:
No clue, this is code that was just moved from the old singleton class!

Reply:
Yeah, my bad. This is stuff for later!